sympy/functions/elementary/hyperbolic.py

```diff
@@ -1359,6 +1380,14 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
     def _eval_rewrite_as_log(self, x, **kwargs):
         return log(x + sqrt(x + 1) * sqrt(x - 1))
 
+    def _eval_rewrite_as_asinh(self, x, **kwargs):
+        return sqrt(x - 1)/sqrt(1 - x)*(pi/2 +
+                                        S.ImaginaryUnit*asinh(S.ImaginaryUnit*x))
+
+    def _eval_rewrite_as_atanh(self, x, **kwargs):
+        return (pi/2*sqrt(x - 1)/sqrt(1-x)*(1 - x * sqrt(1/x**2)) +
+                sqrt(x - 1)*sqrt(x + 1)/sqrt(x**2 - 1)*atanh(sqrt(x**2 - 1)/x))
```

Comment:
This doesn't always work:
```python
In [7]: f = acosh(x)

In [8]: f.subs(x, -1+I).n()
Out[8]: 1.06127506190504 + 2.23703575928741⋅ⅈ

In [9]: f.rewrite(atanh).subs(x, -1+I).n()
Out[9]: 1.06127506190504 - 0.904556894302381⋅ⅈ
```
(a regression compared to master)

Reply:
Same as this https://functions.wolfram.com/ElementaryFunctions/ArcCosh/27/02/09/01/01/0006/

Reply:
Also here the problem is `subs`:
```python
In [8]: f = acosh(x)

In [9]: f.evalf(subs={x:-1-I})
Out[9]: 1.06127506190504 - 2.23703575928741⋅ⅈ

In [10]: f.rewrite(atanh).evalf(subs={x:-1-I})
Out[10]: 1.06127506190504 - 2.23703575928741⋅ⅈ
```